autograder/autograder/autograding_utils.py

```diff
@@ -120,7 +120,23 @@ def log_stack_trace(log_path, job_id="UNKNOWN", is_batch=False, which_untrusted=
             print("Could not gain a lock on the log file.")
 
 
+def log_container_meta(log_path, name="", container="", event="", time=0):
+    """ Given a log file, create or append container meta data to a log file. """
 
+    now = dateutils.get_current_time()
+    datefile = datetime.strftime(now, "%Y%m%d")+".txt"
+    easy_to_read_date = dateutils.write_submitty_date(now, True)
+    time_unit = "sec"
+    with open(log_path, 'a') as myfile:
+        try:
+            fcntl.flock(myfile,fcntl.LOCK_EX | fcntl.LOCK_NB)
+            print("%s | %s | %s | %s | %.3lf | %3s"
+                  % (easy_to_read_date, name, container,
+                     event, time, time_unit),
+                  file=myfile)
+            fcntl.flock(myfile, fcntl.LOCK_UN)
+        except:
+            print("Could not gain a lock on the log file.")
```

Comment:
This is the third time this block has appeared in this file. Move this into a function and have the three functions that do this call that:
```
def write_to_log(log_path, message):
    with open(log_path, 'a') as log_file:
        try:
            fcntl.flock(log_file, fcntl.LOCK_EX | fcntl.LOCK_NB)
            print(message, file=log_file)
            fcntl.flock(log_file, fcntl.LOCK_UN)
        except:
            print("Could not gain a lock on the log file.")
 ```

Also, let's use a combination of [f-strings](https://realpython.com/python-f-strings/) and list concatenations to simplify creating the message, removing/adding new parameters, etc
```
parts = (easy_to_read_date, name, container, event, f"{time:.3f}", f"{time_unit:3s}")
message = ' | '.join((str(x) for x in parts))
```

You can also move your variable assignments into this tuple creation (and inside the f-string expressions), like `f"{dateutils.write_submitty_date(now, True)}"`.

I'd also probably make it so that the `write_to_log` method accepts the parts array and then does the concatenation of it together (casting anything to a string as necessary), but you can choose to do it wherever.

Reply:
That's a great suggestion! I decided on keeping most of the existing code before tuple creation because I find that putting pythons version of the ternary operator in tuple creation made the code harder to read.